### PR TITLE
CORTX-30048: CSM Logs

### PR DESCRIPTION
**Pull request by ShivamVerma380**
# Pull Request
## Problem Statement
-  CSM| Log improvement
-  JIRA Refrence: CORTX-30048

## Design
-   For Bug, Describe the fix here.
-   For Feature, Post the link for design

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM
Build Status:
custom-ci-build:
image

Logs:
image
Create CSM User and Fetch CSM Users:
image
Get Users By UserId:
image
Delete Users By UserId:
image
Request Id in SessionLogs:
image

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: csm/core/controllers/storage_capacity.py [hunk 1]
```diff
@@ -109,6 +117,39 @@ class S3CapacityView(CsmView):
...
+        except ValidationError as val_err:
+            raise InvalidRequest(f"{ValidationErrorFormatter.format(val_err)}")
+        with ServiceError.guard_service():
+            response = await self._service.get_usage(resource, resource_id)
+            Log.info(
+            f"[{self.request.request_id}] Processed request: {self.request.method} {self.request.path}"\
+            f" User: {self.request.session.credentials.user_id}")
+            return CsmResponse(response)
+            return CsmResponse(response)
+
```
Comment (rohitkolapkar) [inline csm/core/controllers/storage_capacity.py:145]: please fix this duplication.

File: csm/core/agent/api.py [hunk 2]
```diff
@@ -682,4 +693,4 @@ async def _async_push(msg):
     def push(alert):
         coro = CsmRestApi._async_push(alert)
         asyncio.run_coroutine_threadsafe(coro, CsmRestApi._app.loop)
-        return True
+        return True
```
Comment (rohitkolapkar) [inline csm/core/agent/api.py:696]: Add new line at the end of the file

File: csm/core/controllers/login.py [hunk 3]
```diff
@@ -60,10 +60,10 @@ class LogoutView(CsmView):
...
-            f"Processing request: {self.request.method} {self.request.path}"\
+            f"[{self.request.request_id}] Processing request: {self.request.method} {self.request.path}"\
             f" User: {username}")
         session_id = self.request.session.session_id
         await self.request.app.login_service.logout(session_id)
         # TODO: Stop any websocket connection corresponding to this session
-        Log.info(f"Logout successful. User: {username}")
-        return CsmResponse()
+        Log.info(f"[{self.request.request_id}] Logout successful. User: {username}")
+        return CsmResponse()
```
Comment (rohitkolapkar) [inline csm/core/controllers/login.py:69]: Add new line at the end of the file

File: csm/core/controllers/rgw/s3/users.py [hunk 4]
```diff
@@ -430,11 +430,11 @@ async def put(self):
...
+        Log.debug(f"[{self.request.request_id}] Handling iam user quota PUT request"
                   f" path params/request body: {request_body}")
         with ServiceError.guard_service():
             response = await self._service.set_user_quota(**request_body)
             Log.info(
-                f"Processed request: {self.request.method} {self.request.path}"\
+                f"[{self.request.request_id}] Processed request: {self.request.method} {self.request.path}"\
                 f" User: {self.request.session.credentials.user_id}")
-            return CsmResponse(response)
+            return CsmResponse(response)
```
Comment (rohitkolapkar) [inline csm/core/controllers/rgw/s3/users.py:440]: Add new line at the end of the file

File: csm/core/controllers/users.py [hunk 5]
```diff
@@ -196,6 +196,6 @@ async def patch(self):
         resp = await self._service.update_user(
             user_id, user_body, loggedin_user_id)
         Log.info(
-            f"Processed request: {self.request.method} {self.request.path}"\
+            f"[{self.request.request_id}] Processed request: {self.request.method} {self.request.path}"\
             f" User: {loggedin_user_id}")
-        return resp
+        return resp
```
Comment (rohitkolapkar) [inline csm/core/controllers/users.py:201]: Add new line at the end of the file

File: csm/core/services/sessions.py [hunk 6]
```diff
@@ -432,4 +432,4 @@ async def _is_valid_session_exists(self, user_id: str) -> None:
                 await self.update_session_expiry_time(each_session)
                 # return valid session
                 return each_session
-        return None
+        return None
```
Comment (rohitkolapkar) [inline csm/core/services/sessions.py:435]: Add new line at the end of the file

File: csm/core/services/storage_capacity.py [hunk 7]
```diff
@@ -180,9 +180,11 @@ async def get_usage(self, resource, resource_id):
...
         if isinstance(plugin_response, RgwError):
+            Log.error(f"S3ServiceError: {plugin_response.error_code.name}:"\
+                f" {plugin_response.error_message}")
             ServiceError.create(plugin_response)
         users_dict = plugin_response["capacity"]["s3"]["users"]
         users_list = []
         users_list.append(users_dict.copy())
         plugin_response["capacity"]["s3"]["users"] = users_list
-        return plugin_response
+        return plugin_response
```
Comment (rohitkolapkar) [inline csm/core/services/storage_capacity.py:190]: Add new line at the end of the file

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

File: csm/core/controllers/storage_capacity.py (hunk 1 above)
Comment (ShivamVerma380) [reply csm/core/controllers/storage_capacity.py]: Fixed.

File: csm/core/agent/api.py (hunk 2 above)
Comment (ShivamVerma380) [reply csm/core/agent/api.py]: Added new line at end of file

File: csm/core/controllers/login.py (hunk 3 above)
Comment (ShivamVerma380) [reply csm/core/controllers/login.py:69]: Added new line at end of file

File: csm/core/controllers/rgw/s3/users.py (hunk 4 above)
Comment (ShivamVerma380) [reply csm/core/controllers/rgw/s3/users.py:440]: Added new line at end of file

File: csm/core/controllers/users.py (hunk 5 above)
Comment (ShivamVerma380) [reply csm/core/controllers/users.py:201]: Added new line at end of file

File: csm/core/services/sessions.py (hunk 6 above)
Comment (ShivamVerma380) [reply csm/core/services/sessions.py]: Added new line at the end of file

File: csm/core/services/storage_capacity.py (hunk 7 above)
Comment (ShivamVerma380) [reply csm/core/services/storage_capacity.py:190]: Added new line at end of file

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by rohitkolapkar — APPROVED** [review]

**Pull request merged**